packages/python/plotly/plotly/express/_core.py

```diff
@@ -754,6 +754,90 @@ def apply_default_cascade(args):
         args["marginal_x"] = None
 
 
+def build_or_augment_dataframe(args, attrables, array_attrables, constructor):
+    """
+    Constructs an implicit dataframe and modifies `args` in-place.
+    `attrables` is a list of keys into `args`, all of whose corresponding
+    values are converted into columns of a dataframe.
+    Used to be support calls to plotting function that elide a dataframe
+    argument; for example `scatter(x=[1,2], y=[3,4])`.
+    """
+
+    # We start from an empty DataFrame except for the case of functions which
+    # implicitely need all dimensions: Splom, Parcats, Parcoords
+    # This could be refined when dimensions is given
+    if constructor in [go.Splom, go.Parcats, go.Parcoords]:  # we take all dimensions
+        df = args["data_frame"]
+    else:
+        df = pd.DataFrame()
+
+    # Retrieve labels (to change column names)
+    labels = args.get("labels")  # labels or None
```

Comment:
is it important to move the (re-)labelling logic into this function? Right now it all mostly happens later with various helper functions. If we're moving this logic into this function then should we refactor it out of where it is right now...?

Reply:
yes, it's better to leave the labelling logic later as it was before, this should be resolved now.